ctypesgen/printer_python/printer.py

```diff
@@ -297,61 +298,73 @@ def print_function(self, function):
         else:
             self.print_fixed_function(function)
 
-    def print_fixed_function(self, function):
+    def print_fixed_function(self, function : FunctionDescription):
         self.srcinfo(function.src)
 
         CC = "stdcall" if function.attrib.get("stdcall", False) else "cdecl"
-
-        # If we know what library the function lives in, or we have only a single library,
-        # look there. Otherwise, check all the libraries.
-        use_single_lib = function.source_library or len(self.options.libraries) == 1
-        if use_single_lib:
-            lib = function.source_library if function.source_library else self.options.libraries[0]
+        L = function.source_library
```

Comment:
This is an undesired revert of commit deb059ee499c6c2335922afc1e7159013e017c9d.

Reply:
If you aren't happy with style, an alternative to the `use_single_lib` var might be
```python
if not function.source_library and len(self.options.libraries) == 1:
    function.source_library = self.options.libraries[0]
```
We'd have to be aware of the side effect of mutating the FunctionDescription object, though.
After this were run, the info if the library loader actually found the function or not would be lost.

Reply:
A independent amendment to deb059ee499c6c2335922afc1e7159013e017c9d might be to allow `libraries[0]` only if library loading was truly skipped.
Anyway, instead of falling back to the runtime loop in that case, we should probably handle this directly in `find_source_libraries()` by raising an error if all libraries were loaded but a symbol's origin could not be resolved, because that suggests mismatched binary/headers, which would be ABI unsafe and a caller mistake.

Reply:
> This is an undesired revert of [deb059e](https://github.com/ctypesgen/ctypesgen/commit/deb059ee499c6c2335922afc1e7159013e017c9d)/#174.

This should be fixed now.



> independent amendment to [deb059e](https://github.com/ctypesgen/ctypesgen/commit/deb059ee499c6c2335922afc1e7159013e017c9d) might be to allow `libraries[0]` only if library loading was truly skipped.
> Anyway, instead of falling back to the runtime loop in that case, we should probably handle this directly in `find_source_libraries()` by raising an error if all libraries were loaded but a symbol's origin could not be resolved, because that suggests mismatched binary/headers, which would be ABI unsafe and a caller mistake.

This is should probably be its own PR / issue

Reply:
OK, yeah.

Reply:
nit: part of the comment still got lost, though ("or we have only a single library")